src/Compilers/VisualBasic/Portable/Symbols/Source/SourceMemberContainerTypeSymbol.vb

```diff
@@ -3808,6 +3816,29 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
 
         End Sub
 
+        Private Sub CheckTupleNamesAgainstBase(interfaces As ImmutableHashSet(Of NamedTypeSymbol), diagnostics As DiagnosticBag)
+            Dim baseType As NamedTypeSymbol = Me.BaseTypeNoUseSiteDiagnostics
+            If baseType Is Nothing Then
+                Return
+            End If
+
+            Dim baseInterfaces = New Dictionary(Of NamedTypeSymbol, NamedTypeSymbol)(EqualsIgnoringComparer.InstanceIgnoringTupleNames)
```

Comment:
>Dim baseInterfaces = New Dictionary(Of NamedTypeSymbol, NamedTypeSymbol)(EqualsIgnoringComparer.InstanceIgnoringTupleNames) [](start = 12, length = 123)

Would it make sense to avoid allocating the Dictionary when baseType.AllInterfacesNoUseSiteDiagnostics returns an empty array?